tcms/rpc/tests/test_testexecutionstatus.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=attribute-defined-outside-init, invalid-name, objects-update-used
+
+from xmlrpc.client import ProtocolError
+from tcms.testruns.models import TestExecutionStatus
+from tcms.rpc.tests.utils import APITestCase, APIPermissionsTestCase
+
+
+class TestFilter(APITestCase):
+    """Test TestExecutionStatus.filter"""
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+    def test_filter_statuses(self):
+        execution_status_weights = []
+        for execution_status in TestExecutionStatus.objects.all():
+            if (execution_status.weight < 0):
+                execution_status_weights.append(execution_status.weight)
+
+        self.assertTrue(len(execution_status_weights) > 0)
+
+
+class TestFilterPermission(APIPermissionsTestCase):
+    """Test permission for TestExecutionStatus.filter"""
+
+    permission_label = 'testruns.view_testexecutionstatus'
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+    def verify_api_with_permission(self):
+        execution_status_weights = self.rpc_client.TestExecutionStatus.filter({'weight__lt': 0})
+        self.assertTrue(len(execution_status_weights) > 0)
```

Comment:
Python has a very rich list of assertion methods, use them
https://docs.python.org/3/library/unittest.html#assert-methods

Reply:
got it, I'll take a look and make changes accordingly.

Reply:
would using `assertIsNotNone(execution_status_weights)` make more sense?

Reply:
To my mind the only two alternatives are either `assertIsNotNone()` or `assertGreater()`.

Reply:
To answer the question which assertion method to use answer the question "What are you trying to check for ?" Multiple/all methods may produce the same results (essentially all of them perform assertEqual)  but you want the code to be explicit and quite obvious as to what it does.  As of your current implementation it looks like `execution_status_weights` is a boolean value, which it isn't. 

It is a list and you are interested it checking whether or not this list is empty. Which means it's length is greater than zero!

Reply:
hmm, I think then `assertGreater(len(execution_status_weights), 0)` makes the most sense then.